lib/core/src/Cardano/Wallet/Api/Types.hs

```diff
@@ -1311,6 +1355,81 @@ instance (PassphraseMaxLength purpose, PassphraseMinLength purpose)
 instance ToJSON (ApiT (Passphrase purpose)) where
     toJSON = toJSON . toText . getApiT
 
+instance FromJSON ApiScript where
+    parseJSON =  withObject "Script" $ \o -> do
+        script' <- o .: "script"
+        (ApiScript . ApiT) <$> parseJSON script'
+instance ToJSON ApiScript where
+    toJSON (ApiScript (ApiT script')) =
+        object ["script" .= toJSON script']
+
+instance FromJSON ApiPubKey where
+    parseJSON = parseJSON >=> eitherToParser . first ShowFmt . fromText
+instance ToJSON ApiPubKey where
+    toJSON (ApiPubKey key') = do
+        let hrp = [Bech32.humanReadablePart|addr_vk|]
+        String $ T.decodeUtf8 $ encode (EBech32 hrp) key'
+
+instance FromJSON ApiCredential where
+    parseJSON obj = do
+        key' <-
+            (withObject "ApiCredential" $
+             \o -> o .:? "pub_key" :: Aeson.Parser (Maybe Text)) obj
+        case key' of
```

Comment:
You can more succinctly define this parser with `<|>` and applicative style.

Reply:
missed a point why use here alternative :thinking:

Reply:
I pushed a change to show why. It makes the parser shorter and clearer I think. Alternative means that it tries the parsers in turn from left to right until one succeeds.